nordicNRPlus/processNrplusMessagesAndUpdateThingShadow.ts

```diff
@@ -0,0 +1,41 @@
+import type { Static } from '@sinclair/typebox'
+import type { inputSchemaLwm2mMessage } from '../lambda/webhookHandler.ts'
+import { processMessage } from './processMessage.ts'
+
+export const processNrplusMessagesAndUpdateThingShadow =
+	({
+		ensureThing,
+		updateShadow,
+		log,
+	}: {
+		ensureThing: (thingName: string) => Promise<void>
+		updateShadow: (thingName: string, message: any) => Promise<void>
+		log?: typeof console.log
+	}) =>
+	async (
+		validatedInput: Static<typeof inputSchemaLwm2mMessage>,
+	): Promise<void> => {
+		for (const message of validatedInput.messages) {
+			const { teamId, deviceId } = message
+			const thingName = `${teamId}-${deviceId}`
+			try {
+				await ensureThing(thingName)
```

Comment:
Do not call this for every message, remember if the thing exists.

Reply:
https://github.com/NordicPlayground/thingy-rocks-cloud-aws-js/pull/18/commits/ef4d377b58d6a3e548942f11471e70e9bea6a140